src/uct/ib/base/ib_iface.c

```diff
@@ -841,19 +841,14 @@ static int uct_ib_iface_gid_extract_flid(const union ibv_gid *gid)
     return ntohs(*((uint16_t*)UCS_PTR_BYTE_OFFSET(gid->raw, 4)));
 }
 
-static int uct_ib_iface_is_flid_enabled(uct_ib_iface_t *iface)
-{
-    return iface->config.flid_enabled &&
-           (uct_ib_iface_gid_extract_flid(&iface->gid_info.gid) != 0);
```

Comment:
why is this check removed?

Reply:
uct_ib_iface_is_flid_enabled is not needed anymore since in uct_ib_iface_dev_addr_is_reachable the code was refactored to print the reason, and in uct_ib_iface_resolve_remote_flid() function uct_ib_iface_gid_extract_flid is already checked in the end.

Reply:
`uct_ib_iface_resolve_remote_flid` checks for flid in the remote gid, but you removed check for local gid in iface

Reply:
fixed